src/whisper_at_train/run_combined_training.sh

```
@@ -7,58 +7,124 @@
+
+# Environment setup
+if [[ -n "${SLURM_JOB_ID:-}" ]]; then
+    echo "Running in SLURM environment (Job ID: ${SLURM_JOB_ID})"
+else
+    echo "Running in local environment"
+fi
+
+# Activate virtual environment
+VENV_PATH="${HOME}/.cache/pypoetry/virtualenvs/whisper-at-z6hdRBdT-py3.10/bin/activate"
+if [[ -f "$VENV_PATH" ]]; then
+    source "$VENV_PATH"
```

Shellcheck (0.10.0)

[warning] ShellCheck can't follow non-constant source. Use a directive to specify location.

(SC1090)